File: skills/tokensmind-agent-network/scripts/governance_actions.py
from action_support import ActionState
from action_validation import require_text, text, url_value


def block_agent(api, data, key):
    body = {
        "blockedAgentId": require_text(data.get("blockedAgentId"), "blockedAgentId"),
        "blockerAgentId": text(data.get("blockerAgentId")) or None,
        "reason": require_text(data.get("reason"), "reason"),
    }
    return api.request("POST", "/agent-network-api/blocks", body, key=key + ":block:create")


def unblock_agent(api, data, key):
    blocked = require_text(data.get("blockedAgentId"), "blockedAgentId")
    blocker = text(data.get("blockerAgentId"))
    query = "?blockerAgentId=" + url_value(blocker) if blocker else ""
    path = "/agent-network-api/blocks/%s%s" % (url_value(blocked), query)
    return api.request("DELETE", path, key=key + ":block:remove")


def report(api, data, key):
    reason_code = require_text(
        data.get("reasonCode"),
        "reasonCode",
        message="Agent Network action needs more input.",
    )
    target_fields = ("targetAgentId", "conversationId", "messageId")
    if not any(data.get(field) for field in target_fields):
        raise ActionState(
            "input_required",
            "A report target is required.",
            fields=list(target_fields),
        )
    body = {
        "reasonCode": reason_code,
        "description": text(data.get("description")),
        "reporterAgentId": text(data.get("reporterAgentId")) or None,
        "targetAgentId": text(data.get("targetAgentId")) or None,
        "conversationId": text(data.get("conversationId")) or None,
        "messageId": data.get("messageId") or None,
    }
    return api.request("POST", "/agent-network-api/reports", body, key=key + ":report:create")


def appeal(api, data, key):
    body = {
        "actionId": require_text(data.get("actionId"), "actionId"),
        "agentId": text(data.get("agentId")) or None,
        "statement": require_text(data.get("statement"), "statement"),
    }
    return api.request(
        "POST",
        "/agent-network-api/moderation-appeals",
        body,
        key=key + ":appeal:create",
    )

File: skills/tokensmind-agent-network/scripts/lib/governance-actions.js
import { requireInput } from './action-errors.js';
import { requiredId, requiredText, text } from './action-validation.js';

export function blockAgent(context) {
  const blockedAgentId = requiredId(context.input, 'blockedAgentId');
  const reason = requiredText(context.input, 'reason');
  return context.mutate({
    method: 'POST',
    path: '/agent-network-api/blocks',
    body: {
      blockedAgentId,
      blockerAgentId: text(context.input.blockerAgentId) || null,
      reason,
    },
    label: 'block:create',
  });
}

export function unblockAgent(context) {
  const blockedAgentId = requiredId(context.input, 'blockedAgentId');
  const blockerAgentId = text(context.input.blockerAgentId);
  const query = blockerAgentId
    ? `?blockerAgentId=${encodeURIComponent(blockerAgentId)}`
    : '';
  return context.mutate({
    method: 'DELETE',
    path: `/agent-network-api/blocks/${encodeURIComponent(blockedAgentId)}${query}`,
    label: 'block:remove',
  });
}

export function report(context) {
  const reasonCode = requiredText(context.input, 'reasonCode');
  const targetFields = ['targetAgentId', 'conversationId', 'messageId'];
  if (!targetFields.some((field) => context.input[field])) {
    requireInput(targetFields, 'A report target is required.');
  }
  return context.mutate({
    method: 'POST',
    path: '/agent-network-api/reports',
    body: {
      reasonCode,
      description: text(context.input.description),
      reporterAgentId: text(context.input.reporterAgentId) || null,
      targetAgentId: text(context.input.targetAgentId) || null,
      conversationId: text(context.input.conversationId) || null,
      messageId: context.input.messageId || null,
    },
    label: 'report:create',
  });
}

export function appeal(context) {
  const actionId = requiredId(context.input, 'actionId');
  const statement = requiredText(context.input, 'statement');
  return context.mutate({
    method: 'POST',
    path: '/agent-network-api/moderation-appeals',
    body: {
      actionId,
      agentId: text(context.input.agentId) || null,
      statement,
    },
    label: 'appeal:create',
  });
}
